mbus/client.go

```diff
@@ -54,10 +55,24 @@ func Connect(c *config.Config, reconnected chan<- Signal, l logger.Logger) *nats
 }
 
 func natsOptions(l logger.Logger, c *config.Config, natsHost *atomic.Value, reconnected chan<- Signal) nats.Options {
-	natsServers := c.NatsServers()
-
 	options := nats.DefaultOptions
-	options.Servers = natsServers
+	options.Servers = c.NatsServers()
+	if c.Nats.TLSEnabled {
+		var err error
+		options.TLSConfig, err = tlsconfig.Build(
+			tlsconfig.WithInternalServiceDefaults(),
+			tlsconfig.WithIdentity(c.Nats.ClientAuthCertificate),
+		).Client(
+			tlsconfig.WithAuthority(c.Nats.CAPool),
+		)
+		// FIXME: Improve this error handling
```

Comment:
Hi @46bit, 

❓ Can you either (1) improve this error handling, (2) explain how you think this error handling could be improved, or (3) explain how you think remove this comment

Reply:
Oh no! I'll fix this next week 🙈

Reply:
Had a look :) I've removed the comment but kept the fatal erroring. This function is only called by the function above it, and that function treats errors as fatal. I think that makes sense–if this function fails then there's something wrong with Gorouter's config and retrying won't help.